src/components/Widgets/DateTimeControl.js

```diff
@@ -1,20 +1,50 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import DateTime from 'react-datetime';
+import ImmutablePropTypes from 'react-immutable-proptypes';
 
 export default class DateTimeControl extends React.Component {
-  componentDidMount() {
-    if (!this.props.value) {
-      this.props.onChange(new Date());
-    }
+  constructor(props) {
+    super(props);
+    this.state = { value: this.getStateValue(this.props.value) };
   }
 
   handleChange = (datetime) => {
     this.props.onChange(datetime);
   };
 
+  getStateValue(value) {
+    if (value === false || value === '') {
+      return '';
+    } else if(!value) {
+      return new Date();
+    } else {
+      return value;
+    }
+  }
+
+  componentWillReceiveProps(props) {
+    const { value: newValue } = props;
+    if(newValue !== this.props.value) {
+      this.setState({ value: this.getStateValue(newValue) }, () => {
+        this.handleChange(this.state.value);
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.handleChange(this.state.value);
+  }
+
   render() {
-    return <DateTime value={this.props.value} onChange={this.handleChange} />;
+    const _fieldProps = this.props.field.get('field');
+    const fieldProps = _fieldProps && _fieldProps.toJSON ? _fieldProps.toJSON() : _fieldProps
```

Comment:
We can condense these two lines:

`const fieldProps = this.props.field.get('field', Map()).toJS()`